modules/xray/shared/src/main/scala/natchez/xray/XRayEnvironment.scala

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2019-2020 by Rob Norris and Contributors
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez.xray
+
+import cats.data._
+import cats.effect._
+import com.comcast.ip4s.{IpAddress, SocketAddress}
+import natchez.Kernel
+
+trait XRayEnvironment[F[_]] {
+  def daemonAddress: F[Option[SocketAddress[IpAddress]]]
+  def traceId: F[Option[String]]
+  def kernelFromEnvironment: F[Kernel]
+}
+
+object XRayEnvironment {
+  def apply[F[_] : XRayEnvironment]: XRayEnvironment[F] = implicitly
+
+  implicit def instance[F[_] : Sync]: XRayEnvironment[F] = new XRayEnvironment[F] {
+    override def daemonAddress: F[Option[SocketAddress[IpAddress]]] =
+      OptionT(Sync[F].delay(sys.env.get("AWS_XRAY_DAEMON_ADDRESS")))
```

Comment:
Unfortunately `sys.env` is not cross-platform, here are some example shims:
https://github.com/armanbilge/gcp4s/blob/main/core/jvm/src/main/scala/gcp4s/platform.scala
https://github.com/armanbilge/gcp4s/blob/main/core/js/src/main/scala/gcp4s/platform.scala

Reply:
Oh, bummer. I introduced similar shims here. Thanks for letting us know!